### ISSUE TITLE
Migration to CMake

### ISSUE THREAD

**Issue by pirogronian** [CelestiaProject/Celestia#131]
I know, that migration to cmake and removing qmake support is one of milestones of version 1.7. So, what is a status of cmake integration? What else could be done?

**Comment by 375gnu** [comment]
Qmake removal wasn't planned for 1.7. I'd say for 1.8.

When built using cmake Celestia crashes on exit in free() function. I haven't find a root cause yet, so if I'm unable to find them, I'd postpone cmake as well (or at least postpone making it default).

**Comment by GottfriedSp** [comment]
Why use now cmake and not qt-build system with qmake?
cmake does not work properly with windows. I installed vcpkg and every time i install something with it, I must disable virus protection, otherwise the CMakeCCompilerid.exe runs into quarantine.

**Comment by pirogronian** [comment]
@GottfriedSp I'd say it's problem of virus protection, not necessarily `vcpkg`...

**Comment by 375gnu** [comment]
@GottfriedSp, install a better antivirus, mine is Devuan GNU/Linux, you should definitely try it.

**Comment by GottfriedSp** [comment]
I have windows and don't like command line tools. Other antivirus I tried, but the most of them slows down the entire system. I also have Debian Linux, but I don't like it. When update my raspberry pi to new debian, some of the functions I had doesn't work anymore, other things changed too much, other things removed. GNOME was changed to 3.0 and does not work like before. On other machines, no support of some hardware and so on. Ubuntu changed the desktop to Unity. To remove the Unity and change to Gnome was terrible and didn't work properly. So I removed teh Ubuntu systems from my computers and never installed them again.

Visual studio projects are much more better than CMake.
With CMake the projects in windows do not add relative path, they generated all with absolute path.
When changing some project parameters, CMake reverted all parameters back.
CMake do only make one target (32-bit OR 64-bit) and can't handle both in one project.
With Visual Studio project, I can build for 32-bit and 64-bit and can test both binaries at the same time without rebuild for each platforms.
qmake projects can also handle 32-bit and 64-bit with qt-creator and can be easy ported into Visual Studio project.

There is also a Qbs System which can handle qt-creator, but this is realy alpha. It detects automatically all dependencies and compiles only changed files and their dependency. This makes building realy fast and saves time when developing (no rebuild is required).

I tried some IDEs at Linux: Code::Blocks, Eclipse, CodeLite, KDevelop and qt-creator.
None of them was my favorite.
CodeLite looks good, it was fast on raspberry pi, but now it was removed from last debian release.
The others are too heavy on raspberry and other slower machines.
qt-creator does not work via remote desktop or virtual machine.

**Comment by 375gnu** [comment]
On 2/8/19, GottfriedSp <notifications@github.com> wrote:
> I have windows and don't like command line tools. Other antivirus I tried,
> but the most of them slows down the entire system. I also have Debian Linux,
> but I don't like it. When update my raspberry pi to new debian, some of the
> functions I had doesn't work anymore, other things changed too much, other
> things removed. GNOME was changed to 3.0 and does not work like before. On
> other machines, no support of some hardware and so on. Ubuntu changed the

I run Debian (now Devuan actually) since 2008 and only once I had to
reinstall when I migrated from 32 to 64 bits. But my configuration is
the same, but I don't use neither Gnome nor KDE.

> Visual studio projects are much more better than CMake.

CMake works on variety of systems. VS works only on Woe. That's enough
to choose CMake over VS projects. I evaluated a lot of build systems.
CMake is far from being perfect, but at least it fits all our needs.

> With CMake the projects in windows do not add relative path, they generated
> all with absolute path.

CMake can generate variety of build configs, I tested Ninja on Woe for example.

> When changing some project parameters, CMake reverted all parameters back.

Don't change intermediate files. Change CMakeLists.txt.

> CMake do only make one target (32-bit OR 64-bit) and can't handle both in
> one project.

And that's how it works on all systems.

> There is also a Qbs System which can handle qt-creator, but this is realy
> alpha. It detects automatically all dependencies and compiles only changed
> files and their dependency. This makes building realy fast and saves time
> when developing (no rebuild is required).

Qbs is going to be dropped in Qt 6.0.

> I tried some IDEs at Linux: Code::Blocks, Eclipse, CodeLite, KDevelop and
> qt-creator.
> None of them was my favorite.
> CodeLite looks good, it was fast on raspberry pi, but now it was removed
> from last debian release.
> The others are too heavy on raspberry and other slower machines.
> qt-creator does not work via remote desktop or virtual machine.

Vim and Emacs work just fine. Even on Woe.
